### PR TITLE
Adds secondary build for vulkan_validation_layers as temporary patch.

### PR DESCRIPTION
**Pull request by gaaclarke**
This is required for getting validation happening for vulkan unit tests.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: build/secondary/third_party/vulkan_validation_layers/BUILD.gn [hunk 1]
```diff
@@ -0,0 +1,490 @@
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+################################################################################
+# This is a temporary secondary build file until the rtti issue is solved
+# in vulkan-deps
```
Comment (dnfield) [inline build/secondary/third_party/vulkan_validation_layers/BUILD.gn:7]: Please file an issue tracking removal of this and link it here.

**Review by dnfield — APPROVED** [review]
LGTM

File: build/secondary/third_party/vulkan_validation_layers/BUILD.gn (hunk 1 above)
Comment (gaaclarke) [reply build/secondary/third_party/vulkan_validation_layers/BUILD.gn:7]: Do you want one for flutter?  I'm not familiar with what bug tracker would be managing upstream issues.  There is a link to the fuchsia bug tracker in the code.

Comment (gaaclarke) [reply build/secondary/third_party/vulkan_validation_layers/BUILD.gn:7]: done: https://github.com/flutter/flutter/issues/127624

**Pull request merged**